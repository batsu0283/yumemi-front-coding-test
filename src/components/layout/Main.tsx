import type { ReactNode } from "react";

type MainProps = {
  children: ReactNode;
};

export const Main = ({ children }: MainProps) => {
  return <main className="container mx-auto p-10">{children}</main>;
};
