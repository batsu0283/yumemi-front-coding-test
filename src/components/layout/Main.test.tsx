import { render, screen } from "@testing-library/react";
import { Main } from "./Main";

describe("Main", () => {
  it("子要素が正しくレンダリングされること", () => {
    render(
      <Main>
        <p>テスト用の子要素</p>
      </Main>,
    );
    expect(screen.getByText("テスト用の子要素")).toBeInTheDocument();
  });

  it("mainの要素が存在すること", () => {
    render(
      <Main>
        <p>Another Child</p>
      </Main>,
    );
    expect(screen.getByRole("main")).toBeInTheDocument();
  });

  it("複数の子要素をレンダリングできること", () => {
    render(
      <Main>
        <p>子要素1</p>
        <p>子要素2</p>
      </Main>,
    );

    // 複数の子要素が正しく表示されているか確認
    expect(screen.getByText("子要素1")).toBeInTheDocument();
    expect(screen.getByText("子要素2")).toBeInTheDocument();
  });
  it("Snapshot", () => {
    const { container } = render(
      <Main>
        <p>子要素1</p>
        <p>子要素2</p>
      </Main>,
    );
    expect(container).toMatchSnapshot();
  });
});
