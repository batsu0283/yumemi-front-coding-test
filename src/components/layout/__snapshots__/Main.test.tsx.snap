// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main > Snapshot 1`] = `
<div>
  <main
    class="container mx-auto p-10"
  >
    <p>
      子要素1
    </p>
    <p>
      子要素2
    </p>
  </main>
</div>
`;
